playbooks/roles/caasp4/tasks/setup_nodes.yml

```diff
@@ -55,3 +55,30 @@
   template:
     src: inventory-caasp4.yml.j2
     dest: "{{ socok8s_workspace }}/inventory/caasp4.yml"
+
+- meta: refresh_inventory
+
+- name: Get caasp ips
+  set_fact:
+    ssh_known_hosts_masters: "{{ groups['caasp-masters'] | map('extract', hostvars, ['ansible_host']) | list }}"
```

Comment:
I wonder if we want to pass a variable instead of using the `groups` variable here. Currently **all** variables that are used in the caasp4 role are defined in `defaults/main.yml` and set in `playbooks/openstack-deploy-caasp4.yml`. That way we have a clean interface (the variables) to interact with the role (instead of accessing $random variables from everywhere).

Reply:
but this is not a random variable, its a dynamic variable that can change in teh future (more masters? more workers?)

We need to calculate it on the fly for this to work.

Reply:
Yeah - but it assumes that somebody created a group variable with the specific name . You need to
- a) document it somewhere that a user who wants to reuse this role needs to set it

or

b) Add it to defaults/main.yml (as empty list) and then overwrite it when calling the tasks that need the variable (in `playbooks/openstack-deploy-caasp4.yml`).

Reply:
but that group var comes form the inventory and we are in control of the inventory names no? They should not change at all if we want everything to work, otherwise everything else that refers to caasp-workers/masters will fail to deploy

Reply:
ah. True. the variables are filled from the caasp4 role. sorry for the noise.